kmyth_log/include/kmyth_log.h

```diff
@@ -10,152 +10,53 @@
-                                                   TPM2_RC_SESSION_MEMORY)
+#define DEFAULT_APP_NAME "default_app"
 
 /**
- * @defgroup misc_resmgr_rc TPM 2.0 RCs from TAB/RM
- *
- * @brief These return code (RC) definitons are used to make sense of some
- *        additional response codes that may be received for the TPM 2.0
- *        Access Broker (TAB) / Resource Manager (RM) API.
- *
- * In the tpm2-abrmd-2.0.2 source code, there are several locations where
- * the resource manager simply returns a single or combination of TPM 2.0
- * return codes cast to type RM_RC (resource manager RC).
+ * @brief maximum length (in chars) of log name
  */
+#define MAX_APP_NAME_LEN 32
 
-/** 
- * @ingroup misc_resmgr_rc
- *
- * @brief   Memory error code (RM_RC (TPM2_RC_MEMORY))
+/**
+ * @brief default version string identifying application version being logged
  *
- * Found in tpm2-abrmd-2.0.2/src/access-broker.c, and is returned when
- * allocating a TPM2 response buffer fails.
+ * Note: this default value (and any user specified value) must comply with
+ *       the string length restrictions imposed by MAX_APP_VERSION_LEN (below)
  */
-#define RESMGR_TPM2_RC_MEMORY           (TSS2_RC) TSS2_RESMGR_TPM_RC_LAYER + \
-                                        TPM2_RC_MEMORY
+#define DEFAULT_APP_VERSION "0.0.0"
 
-/** 
- * @ingroup misc_resmgr_rc
- *
- * @brief   Insufficient error code (RM_RC (TPM2_RC_INSUFFICIENT))
- *
- * Found in tpm2-abrmd-2.0.2/src/tpm2-command.c, and is returned when the
- * command buffer's size is insufficient.
+/**
+ * @brief maximum length (in chars) of application version string
  */
-#define RESMGR_TPM2_RC_INSUFFICIENT     (TSS2_RC) TSS2_RESMGR_TPM_RC_LAYER + \
-                                       TPM2_RC_INSUFFICIENT
+#define MAX_APP_VERSION_LEN 16
 
-/** 
- * @ingroup misc_resmgr_rc
- *
- * @brief   Type error code (RM_RC (TPM2_RC_TYPE))
+/**
+ * @brief default path string for application log file
  *
- * Found in tpm2-abrmd-2.0.2/src/tpm2-command.c, and is returned when
- * tpm2_command_get_flush_handle is called with the wrong command.
+ * Note: this default value (and any user specified value)  must comply with
+ *       the string length restrictions imposed by MAX_APPLOG_PATH_LEN (below)
  */
-#define RESMGR_TPM2_RC_TYPE             (TSS2_RC) TSS2_RESMGR_TPM_RC_LAYER + \
-                                       TPM2_RC_TYPE
+#define DEFAULT_APPLOG_PATH "/var/log/default_app.log"
 
-/** 
- * @ingroup misc_resmgr_rc
- *
- * @brief   Handle error code (RM_RC (TPM2_RC_HANDLE + TPM2_RC_P + TPM2_RC_1))
- *
- * Found in tpm2-abrmd-2.0.2/src/resource-manager.c, and is returned when
- * a handle doesn't map to one that the resource manager is managing.
- * TPM2_RC_HANDLE indicates that the error is related to a handle,
- * TPM2_RC_P signifies that is is a parameter, and
- * TPM2_RC_1 specifies that it is the first parameter.
+/**
+ * @brief maximum length (in chars) of path string for the application log file
```

Comment:
Does this include the terminating null character? This same comment applies to the descriptions of the maximum lengths for app name and version.

Reply:
No, the maximum length parameters (MAX_APP_NAME_LEN, MAX_APP_VERSION_LEN, MAX_APPLOG_PATH_LEN, and MAX_LOG_MSG_LEN) do not include the terminating null character.

I think I verified that everywhere I use this value to specify array sizes for the corresponding string values, I add one to the maximum length value to accommodate for the string terminator.

When formatting the log message string (in log_event()), I pass MAX_LOG_MSG_LEN+1 as the count parameter to vnsprintf(), because the documentation indicates that this should include the terminating null character.

When using strnlen(), I believe that I correctly expect the returned length to not include the terminating null character.